modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/extension/v1_0/test/ObjectEntryExtensionProviderTest.java

```diff
@@ -0,0 +1,231 @@
+			"decimalField", false);
+		_addCustomObjectField(
+			objectDefinition.getObjectDefinitionId(), "PrecisionDecimal",
+			"BigDecimal", "precisionDecimalField", true);
+
+		_user = UserTestUtil.addUser();
+	}
+
+	@AfterClass
+	public static void tearDownClass() throws Exception {
+		PropsUtil.addProperties(
+			UnicodePropertiesBuilder.setProperty(
+				"feature.flag.LPS-135404", "false"
+			).build());
+	}
+
+	@Test
+	public void testGetExtendedPropertyDefinitions() throws Exception {
+		Map<String, PropertyDefinition> extendedPropertyDefinitions =
+			_extensionProvider.getExtendedPropertyDefinitions(
+				TestPropsValues.getCompanyId(), UserAccount.class.getName());
+
+		_assertPropertyDefinition(
+			extendedPropertyDefinitions.get("booleanField"), "booleanField",
+			PropertyDefinition.PropertyType.BOOLEAN, false);
+		_assertPropertyDefinition(
+			extendedPropertyDefinitions.get("dateField"), "dateField",
+			PropertyDefinition.PropertyType.DATE_TIME, true);
+		_assertPropertyDefinition(
+			extendedPropertyDefinitions.get("decimalField"), "decimalField",
+			PropertyDefinition.PropertyType.DOUBLE, false);
+		_assertPropertyDefinition(
+			extendedPropertyDefinitions.get("precisionDecimalField"),
+			"precisionDecimalField",
+			PropertyDefinition.PropertyType.BIG_DECIMAL, true);
+	}
+
+	@Test
+	public void testIsApplicableExtension() throws Exception {
+		Assert.assertFalse(
+			_extensionProvider.isApplicableExtension(
+				TestPropsValues.getCompanyId(), null));
+		Assert.assertTrue(
+			_extensionProvider.isApplicableExtension(
+				TestPropsValues.getCompanyId(), UserAccount.class.getName()));
+	}
+
+	@Test
+	public void testSetAndGetExtendedProperties() throws Exception {
+		UserAccount userAccount = new UserAccount() {
+			{
+				setId(_user.getUserId());
+			}
+		};
+
+		_testSetAndGetExtendedProperties(
+			userAccount,
+			HashMapBuilder.<String, Serializable>put(
+				"booleanField", true
+			).put(
+				"dateField", "2010-12-25"
+			).put(
+				"decimalField", 1.2
+			).put(
+				"precisionDecimalField", 100.5
+			).build());
+		_testSetAndGetExtendedProperties(
+			userAccount,
+			HashMapBuilder.<String, Serializable>put(
+				"booleanField", false
+			).put(
+				"dateField", "2020-07-30"
+			).put(
+				"decimalField", 10.8
+			).put(
+				"precisionDecimalField", 20.55
+			).build());
+	}
+
+	private static void _addCustomObjectField(
+			long objectDefinitionId, String businessType, String dbType,
+			String name, boolean required)
+		throws Exception {
+
+		_objectFieldLocalService.addCustomObjectField(
+			TestPropsValues.getUserId(), 0, objectDefinitionId, businessType,
+			dbType, null, RandomTestUtil.randomBoolean(),
+			RandomTestUtil.randomBoolean(), null,
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			name, required, false, Collections.emptyList());
+	}
+
+	private void _assertPropertyDefinition(
+		PropertyDefinition propertyDefinition, String propertyName,
```

Comment:
Can we change these propertyName, propertyType and required to include the "expected" prefix?

Reply:
I'll change them

Reply:
Done!

Reply:
Thanks 🙏🏻